jbmc/src/java_bytecode/assignments_from_json.cpp

```diff
@@ -257,25 +255,24 @@ static jsont get_untyped_string(const jsont &json)
 static optionalt<java_class_typet> given_runtime_type(
   const jsont &json,
   const optionalt<std::string> &type_from_array,
-  symbol_table_baset &symbol_table)
-{
-  std::string runtime_type;
-  if(!has_type(json) && !type_from_array)
-    return {};
-  if(has_type(json))
-    runtime_type = "java::" + json["@type"].value;
-  else
-  {
-    PRECONDITION(
+  const symbol_table_baset &symbol_table)
+{
+  const auto runtime_type = [&]() -> optionalt<std::string> {
+    if(const auto type_from_json = get_type(json))
+      return "java::" + *type_from_json;
+    if(!type_from_array)
+      return {};
+    INVARIANT(
       type_from_array->find('L') == 0 &&
-      type_from_array->rfind(';') == type_from_array->length() - 1);
-    runtime_type =
-      "java::" + type_from_array->substr(1, type_from_array->length() - 2);
-  }
-  if(!symbol_table.has_symbol(runtime_type))
+        type_from_array->rfind(';') == type_from_array->length() - 1,
+      "Types inferred from the type of a containing array should be of the "
+      "form Lmy.package.name.ClassName;");
+    return "java::" + type_from_array->substr(1, type_from_array->length() - 2);
+  }();
+  if(!runtime_type || !symbol_table.has_symbol(*runtime_type))
```

Comment:
:pick: For code clarity I would put the `!runtime_type` test before the invariant check.

Reply:
Not sure if I understand this comment but `runtime_type` is no longer an optional so this part of the test is gone.